### PR TITLE
Bundler 2: Add native helpers for LockfileUpdater

### PR DESCRIPTION
**Pull request by jurre**
Add native helpers for bundler 2 lockfile updater.

### CODE REVIEW

File: bundler/spec/dependabot/bundler/file_updater_spec.rb [hunk 1]
```diff
@@ -878,6 +881,56 @@
...
+              source: {
+                type: "git",
+                url: "https://github.com/elastic/elasticsearch-ruby.git",
+                branch: "5.x",
+                ref: "5.x"
+              }
+            }]
+          end
+
+          it "updates the dependency's revision" do
```
Comment (feelepxyz) [inline bundler/spec/dependabot/bundler/file_updater_spec.rb:913]: Is this true? 😅 seems like the test is checking if we keep the same revision as the old lockfile? 🤔 

Looks like the requirements are not being updated at all so not clear what's being tested. What happens if we do end up updating the requirement? Do we need to unlock the other gems also?

File: bundler/spec/dependabot/bundler/file_updater_spec.rb [hunk 2]
```diff
@@ -1709,27 +1762,23 @@
...
-      )
-    end
-
+  context "with bundler 2 support enabled", :bundler_v2_only do
     describe "updated_dependency_files" do
-      it "fails as the native helper is not yet implemented" do
-        expect { updater.updated_dependency_files }.
-          to raise_error(Dependabot::NotImplemented,
-                         "Bundler 2 adapter does not yet implement update_lockfile")
+      it "updates the dependencies" do
```
Comment (feelepxyz) [inline bundler/spec/dependabot/bundler/file_updater_spec.rb:1767]: Is this case already tested elsewhere for bundler v2? Could we ✂️ ?

File: bundler/spec/dependabot/bundler/file_updater_spec.rb [hunk 3]
```diff
@@ -1709,27 +1762,23 @@
...
+
+      context "with a dependency that requiers bundler v2" do
+        let(:dependency_files) { project_dependency_files("bundler2/requires_bundler") }
+        let(:dependency_name) { "guard-bundler" }
+        let(:requirements) do
+          [{ file: "Gemfile", requirement: "3.0.0", groups: [], source: nil }]
+        end
+
+        it "updates the version" do
+        end
```
Comment (feelepxyz) [inline bundler/spec/dependabot/bundler/file_updater_spec.rb:1781]: Looks like the expectation is missing

File: bundler/helpers/v2/lib/functions.rb [hunk 4]
```diff
@@ -1,29 +1,40 @@
+# frozen_string_literal: true
```
Comment (feelepxyz) [inline bundler/helpers/v2/lib/functions.rb:1]: Thanks for fixing the rubocops ✨

File: bundler/spec/dependabot/bundler/file_updater_spec.rb (hunk 1 above)
Comment (jurre) [reply bundler/spec/dependabot/bundler/file_updater_spec.rb:913]: This was mostly just adapted from the v1 version, but I think what this is supposed to test is that we keep the same remote after the update, but that it does not have the same revision:

```
            expect(new_remote_line).to eq(original_remote_line)
            expect(new_revision_line).to_not eq(original_revision_line)
```

This seems right to me?

File: bundler/spec/dependabot/bundler/file_updater_spec.rb (hunk 2 above)
Comment (jurre) [reply bundler/spec/dependabot/bundler/file_updater_spec.rb:1767]: I think so, yes! ✨

File: bundler/spec/dependabot/bundler/file_updater_spec.rb (hunk 1 above)
Comment (feelepxyz) [reply bundler/spec/dependabot/bundler/file_updater_spec.rb:913]: Oh yes you're right, totally missed the `to_not eq` bit! 👍

**Review by feelepxyz — APPROVED** [review]
LGTM 👍  think you also need to yank out these specs https://github.com/dependabot/dependabot-core/blob/main/bundler/helpers/v2/spec/functions_spec.rb#L8-L9

File: bundler/spec/dependabot/bundler/file_updater_spec.rb (hunk 1 above)
Comment (jurre) [reply bundler/spec/dependabot/bundler/file_updater_spec.rb:913]: Let's chance it to `not_to` 👍

File: bundler/helpers/v2/lib/functions.rb (hunk 4 above)
Comment (feelepxyz) [inline bundler/helpers/v2/lib/functions.rb:1]: Looks like my merge on main created a merge conflict in this file

**Comment by jurre** [comment]
The python failures seem unrelated, and with that suite taking 60m at the minute (💀), I'm gonna go ahead and merge this. If those tests continue to fail and are not a flake, I'll open a separate PR to fix

**Pull request merged**